sessiontxn/isolation/readcommitted.go

```diff
@@ -77,13 +78,30 @@ func NewPessimisticRCTxnContextProvider(sctx sessionctx.Context, causalConsisten
 }
 
 // OnStmtStart is the hook that should be called when a new statement started
-func (p *PessimisticRCTxnContextProvider) OnStmtStart(ctx context.Context) error {
-	if err := p.baseTxnContextProvider.OnStmtStart(ctx); err != nil {
+func (p *PessimisticRCTxnContextProvider) OnStmtStart(ctx context.Context, node ast.StmtNode) error {
+	if err := p.baseTxnContextProvider.OnStmtStart(ctx, node); err != nil {
 		return err
 	}
+
+	// Try to mark the `RCCheckTS` flag for the first time execution of in-transaction read requests
+	// using read-consistency isolation level.
+	if node != nil && NeedSetRCCheckTSFlag(p.sctx, node) {
+		p.sctx.GetSessionVars().StmtCtx.RCCheckTS = true
```

Comment:
In `session.ExecutePreparedStmt` ,  the `OnStmtStart` will be called before `ResetContextOfStmt` so `RCCheckTS` will always be false

Reply:
Yes, we should move the ResetContextOfStmt.